### DISCUSSION TITLE
to_netcdf adding unwanted information

### DISCUSSION THREAD

**Discussion by HelixPiano** [pydata/xarray#9538] [Q&A]
Hello everyone,
I am currently struggling with the `to_netcdf` routine.
I have several hundred different netcdf files and some of them contain an unwanted attribute.
Furthermore, the files are for different variables, so they might contain various attributes and coordinates.
I run somthing simple like:
```
df=xr.open_dataset("file_here.nc")
df=df.drop_vars("time_bnds")
df.to_netcdf("new_file.nc")
```

My problem is that xarray starts adding new information to the file that were originally not there.
Excerpt of the original file:
![right](https://github.com/user-attachments/assets/78f2bc54-1377-4e0f-ab76-663efd5c7834)

And this is how it looks modified:
![false](https://github.com/user-attachments/assets/ee977985-3198-4e98-864f-36ad782a25ea)

As you can see "_FillValue" was added to `time` and `char rotated pole`  turned into `char roated_pole(string1)` as well as having `rotated poles:coordinates` added.

It is important that everything else (except for the dropped variable) stays the same.
The documentation for `to_netcdf` mentions that I can remove the `_FillValue` by setting it in the encoding to None, but this would require a full dictenory with the entire encoding of every variable; not to mention that there are differences between my files.
Is there any simple way to do that, or do I have to ditch xarray for that one?

**Comment by marceloandrioni** [comment]
Maybe something like this:

```
with xr.open_dataset("file_here.nc") as ds:

    ds = ds.drop_vars("time_bnds")

    ds["rotated_pole"].encoding["dtype"] = "S1"
    ds["time"].encoding["_FillValue"] = None

    ds.to_netcdf("new_file.nc")
```

but without a file for testing I can't be sure. Can you upload an example file here?

**Comment by trexfeathers** [comment]
@HelixPiano have you considered using ncdata to fix your file during save? The [Rename a dimension in xarray output](https://github.com/pp-mo/ncdata?tab=readme-ov-file#rename-a-dimension-in-xarray-output) example demonstrates how this is possible, and you can do a lot more than just renaming dimensions 😊
